### DISCUSSION TITLE
How to set LocalContext for each item in a list

### DISCUSSION THREAD

**Discussion by VincentShaneFinn** [Netflix/dgs-framework#902] [Q&A]
https://netflix.github.io/dgs/advanced/context-passing/
We are using local context to pass internal ids to nested children. There is one use case I haven't seen an example for.

As far as I can tell, Graphql doesn't really distinguish between a single item or a list of items in the schema, and similarly in the DataFetchers.

So, if we had a fetcher that responded with one thing then setting a local context with some internal type is straightforward
```
  @DgsQuery
  public DataFetcherResult<SomeType> getSomeType() {
    var result = service.getSomeType();
    return DataFetcherResult.<SupplierCatalog>newResult()
        .data(result)
        .localContext(result.getInternalResult())
        .build();
  }
```

However, if a Datafetcher returns a list then you can no longer set LocalContext for each individual item, and I don't know if there is a way to have a DateFetcher that resolves a SINGLE ITEM IN THE LIST, to override this context.

```
  @DgsQuery
  public DataFetcherResult<List<SomeType>> getSomeTypes() {
    var result = service.getSomeType();
    //How can I set the local context when branching off to resolve each SomeType's fields and nested fields
    return result;
  }
```

We have ended up having to set the LocalContext for each field for SomeType, and would be nice if we had access to a getSomeType Datafetcher, or something like that.

**Comment by srinivasankavitha** [comment]
In this case, you are probably better off encapsulating the context within a wrapper type that contains `SomeType` fields and fields that capture your local context for each item. Then you can return that as the result, instead of using the DataFetcherResult with context. Something like this: https://netflix.github.io/dgs/advanced/context-passing/#no-showid-use-an-internal-type

Alternatively, you could also do something like this: https://netflix.github.io/dgs/advanced/context-passing/#no-showid-use-local-context

Hope that helps.

**Reply by VincentShaneFinn** [reply]
Well, the "problem" is sort of like we want an Id from the root field to be propogated down to all children and grandchildren, but the root is a list instead of a single item, and the root also has several child fields. It would be nice if there was one place to setup the local context with that wrapper type instead of having to set in each child field of that root. 

Was hoping there might be some way to set the local context "foreach" item in a field that is a list instead of a single item.

**Reply by srinivasankavitha** [reply]
You can set the local context to contain whatever data you want to pass down to child fetchers. Take a look at the `Passing Data and Local Context` section here: https://www.graphql-java.com/blog/deep-dive-data-fetcher-results/

If I understand correctly, that is what you are looking to achieve?
